SPEC.md

```diff
@@ -120,26 +129,28 @@ The following characters cannot be the first character in a
 * Any decimal digit (0-9)
 * Any [non-identifier characters](#non-identifier-characters)
 
-Be aware that the `-` character can only be used as an initial
-character if the second character is not a digit. This allows
-identifiers to look like `--this`, and removes the ambiguity
-of having an identifier look like a negative number.
+Additionally, the `-` character can only be used as an initial character if
+the second character is *not* a digit. This allows identifiers to look like
+`--this`, and removes the ambiguity of having an identifier look like a
+negative number.
 
 ### Non-identifier characters
 
 The following characters cannot be used anywhere in a [Bare Identifier](#identifier):
 
-* Any codepoint with hexadecimal value `0x20` or below.
-* Any codepoint with hexadecimal value higher than `0x10FFFF`.
-* Any of `\/(){}<>;[]=,"`
+* Any of `(){}[]/\="#;`
```

Comment:
Oh, I thought we were just gonna disallow # as the first char, not anywhere.

Reply:
If it was left it the raw string form would be misinterpreted, from what I gathered from conversations is # is going ot be used for keywords (only `#true`, `#false` and `#null` atm)

Reply:
It originally was, but then I thought it might be too confusing to _only_ ban it as the first character. I'm open to banning it only as an initial character, though. Any strong feelings?

Reply:
@IceDragon200 No, `foo#bar` is not ambiguous with anything.

@zkat I don't have *strong* feelings about it, but I'm weakly in favor of allowing it in non-initial positions. Whichever way you prefer is fine, tho.

Reply:
I think I'm going to keep it like this. Tbh if we allow `#` as a non-initial character, we may as well also allow `"/\`, possibly even the bracket types, since once we know an identifier doesn't start with one of these characters, we can unambiguously parse the rest of the ident until the next whitespace. Even for `{`, which now requires a space before it. The question is: at what point does all this "you can use these characters, but _not as first characters_ become too much cognitive load for users? Maybe it's best that we make all our structure-significant tokens completely banned from identifiers?

It also protects about silly mistakes like:

```kdl
node foo#true
```

Thoughts?

Reply:
Yeah I'm fine with this. Your general point is valid (tho we definitely shouldn't ban the numeric characters inside of idents, but I doubt you were intending to refer to those).

Reply:
yeah I'm ok with this change as-is. Thanks